src/test/java/xyz/gvital/app/UserServiceImplTest.java

```diff
@@ -0,0 +1,59 @@
+package xyz.gvital.app;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import xyz.gvital.core.User;
+import xyz.gvital.repo.UserRepository;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceImplTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private UserValidator validator;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    @Test
+    void canRegisterUser() {
+        when(userRepository.saveUser(any(User.class))).thenAnswer(invocation -> {
+            User user = invocation.getArgument(0);
+            user.setId(Math.round(Math.random() * 100));
```

Comment:
Random class could be used
```suggestion
            user.setId(new Random().nextInt(100));
```